### PR TITLE
Completed homework

### PR DESCRIPTION
**Pull request by dabellator**
Hi James,

Let me know if anything needs to be redone.  I tried something funky with the bonus task, it was fun to play around with!

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,4 +1,72 @@
...
+app.use(express.static('public'));
+
+app.get('/trucks', function (req, res) {
+	res.json(trucks.getTrucks());
+});
+
+app.get('/trucks/:name', function (req, res) {
+	var truck = req.params.name;
+	var truckInfo = trucks.getTruck(truck);
+	// Since some of our data is in arrays, it needs to be cleaned up
```
Comment (jamdon) [inline app.js:21]: If I read this correctly, I think you can also just do array.join(' ') to achieve the same effect.

File: public/client.js [hunk 2]
```diff
@@ -1,9 +1,46 @@
...
-// 		});
-// 		$('.truck-list').append(truckList);
-// 	});
-// });
\ No newline at end of file
+$(function () {
+	// Create a function for the truck list so that it can be called
+	// in order to switch between trucks and types.
+	// Immediately invoked to make truck list the default view
+	var truckView = (function truckView() {
```
Comment (jamdon) [inline public/client.js:5]: It looks like you meant to make truckView a function, but what's happening here is truckView is being assigned the result of calling your immediate executed function expression which doesn't explicitly return anything, therefore it returns undefined. If you remove the wrapping IIFE, you'll be able to toggle back and forth between the two views. The issue removing it introduces is that the truckView code isn't automatically called when the script runs, so you'll want to call it once explicitly.

**Comment by jamdon** [comment]
I like the extra flourish of the toggle in the UI to see the trucks vs. the food types. I added a comment inline suggesting a way to make it work better. Additionally, nicely done having the trucks returned from the foodTypes list link back to the individual truck paths as well.:+1:

File: public/client.js (hunk 2 above)
Comment (dabellator) [inline public/client.js:5]: Wow, how did I miss that?  I could have sworn it was working both ways when I was troubleshooting!  I removed the IIFE and just called it before the click event.  Thanks for pointing that out!

File: app.js (hunk 1 above)
Comment (dabellator) [inline app.js:21]: Awesome, that's a good tip.  It's also a good reminder that I should check the available array methods more often, I always forget how many helpers like this there are.

**Pull request closed without merging**